src/Features/CSharp/Portable/InvertIf/CSharpInvertIfCodeRefactoringProvider.cs

```diff
@@ -175,14 +175,19 @@ protected override StatementSyntax AsEmbeddedStatement(IEnumerable<StatementSynt
                     ? SyntaxFactory.Block(trueStatement)
                     : trueStatement);
 
-            if (falseStatementOpt != null)
+            if (falseStatementOpt != null &&
+                ShouleKeepFalse(ifNode, falseStatementOpt))
```

Comment:
maybe just `            if (ShouleKeepFalse(ifNode, falseStatementOpt))` and have ShouldKeepFalse do the null check.

Reply:
I like it